lib/semantics/statements.h

```diff
@@ -0,0 +1,34 @@
+#ifndef FORTRAN_SEMANTICS_STATEMENTS_H_
+#define FORTRAN_SEMANTICS_STATEMENTS_H_
+
+#include "semantics.h"
+#include "../common/indirection.h"
+
+namespace Fortran::parser {
+struct StopStmt;
+}
+
+namespace Fortran::semantics {
+class StatementContext;
+}
+
+extern template class Fortran::common::Indirection<
+    Fortran::semantics::StatementContext>;
+
+namespace Fortran::semantics {
+
+// Semantic analysis of some subset of statements in a parse tree.
+class StatementChecker : public virtual BaseChecker {
+public:
+  explicit StatementChecker(SemanticsContext &);
+  ~StatementChecker();
+
+  void Enter(const parser::StopStmt &);
+
+private:
+  common::Indirection<StatementContext> context_;
```

Comment:
... and removing this, and just keep a reference to the `SemanticsContext` in this checker class.

Reply:
Done.